Added bottom gesture for triggering cursor mover/advanced functions

kugiigi opened the pull request:
## Benefits
Currently, the cursor mover/advanced functions can only be used if and only if, a space key is present in the layout so this excludes layouts such as Japanese, emoji, numpad, etc.
This can be removed and still work but I guess a hint is essential for discoverability and consistency.
This bottom gesture solves this and also improves the flow because instead of pressing and holding and waiting, you just swipe up from the bottom.

## Things to consider
With this change, there's a small but noticeable change in the UI. The standard bottom edge hint is now present at the bottom and the bottom margin has been increased slightly.

It can also be harder to trigger the gesture in landscape mode especially for devices with protruding side edges and those with phone case. Press and hold in the space key is still retained though.

## Screenshots
![screenshot20190925_225723634](https://user-images.githubusercontent.com/29456786/65613290-9cb2c600-df6a-11e9-8c22-004c5557dae2.png)
![screenshot20190925_225739525](https://user-images.githubusercontent.com/29456786/65613291-9d4b5c80-df6a-11e9-81cb-f8b7463afc16.png)
![screenshot20190925_225728905](https://user-images.githubusercontent.com/29456786/65613292-9d4b5c80-df6a-11e9-8d9c-79d4f047960d.png)

Fuseteam commented:
awesome!

UniversalSuperBox commented:
What do you think of this compared to https://github.com/ubports/keyboard-component/issues/103 ?

kugiigi commented:
I like the idea of having the icon in the space bar because it's consistent with the current design however, it only solves the discoverability issue of the finctionality. It does not address the issue with "availability/accessibility". Currently, the advanced features are only accessible with layouts that uses the space key (the normal one and not the small version). For example, it's not accessible in the japanese layout and also the the numpad.

UniversalSuperBox commented:
I've found it quite easy to accidentally move up a few lines while using this change, since the activation happens before releasing your finger. So, the cursor mover starts before I lift my finger after swiping up

kugiigi commented:
Are you sure you're moving up a few lines? or maybe a few characters? Because I actually disabled vertical movements on initial trigger. It'll only be enabled once you re-swipe. But I do agree that maybe there should be more optimization for this. I will revisit this PR in the next days. I've been using it and so far it really makes me use the feature because it's easily triggered and so much faster than pressing and holding.

kugiigi commented:
I've added a delay in actually detecting swipes in cursor swipe mode  which greatly reduce accidental movements after triggering. I think it works a lot better now that I also enabled vertical movement on initial trigger.

I also replaced the usage of `bottom_margin` in some layouts and used `row_margin` to be consistent with the majority of the layouts. Using `bottom_margin` before caused layout issue because I set it to `1` which was `0` before.
